### DISCUSSION TITLE
[24.04] 2024-04-24

### DISCUSSION THREAD

**Discussion by darthcloud** [darthcloud/BlueRetro#979] [General]
# Changelog

## New features
* Add PC-Engine Keyboard support (Thanks @dshadoff)
* Support latest GBAHD OSD
* HW2 Double press on reset shutdown the relay. (Thanks @TharathielCB)
* HW2 Holding reset at power on now send a reset to system. (Thanks @TharathielCB)

## Bug fixes
* Fix GameCube trigger when using multiple controllers simultaneously
* Fix auto detection in universal FW (Thanks @pdaehne)
* Fix FC Keyboard GRPH key
* Fix 8bitdo NeoGeo pad triggers
* Fix Stadia controller mapping
* Fix Stadia controller rumble
* Fix MD/Genesis Mode button mapping
* Fix DC Keyboard random press
* Fix Microsoft mobile 6000 keyboard
* Fix NVS var mapping

# Firmware is available in 2 different flavor
## 1. Regular FW as before retroactively renamed HW1 (Typically for external adapter)
Use firmwares from HW1 archive for:
* [WillConsole's](https://twitter.com/WillConsole) PSUnoRetro
* All [Humble Bazooka's](https://twitter.com/humblebazooka) BT dongle
* [RetroTime's](https://twitter.com/bixxewoscht) N64 dongle
* [RetroOnyx's](https://twitter.com/mellott124) VB dongle
* [GrechTech's](https://twitter.com/GrechTech) RetroRosetta
* any DIY adapter you made base on the original BlueRetro spec
* most likely anything from aliexpress

## 2. HW2 FW that support power management and port detection (Typically for internal install)
Use firmwares from HW2 archive for:
* [Laser Bear Industries's](https://twitter.com/collingall) GameCube BlueRetro Internal Adapter
* any DIY internal install you made base on the HW2 spec

## [OTA Firmware update page](https://blueretro.io/ota.html) has protection to prevent cross flashing between HW1 & HW2 firmware.

# Download latest firmware archives from https://darthcloud.itch.io/blueretro

<hr /><em>This discussion was created from the release <a href='https://github.com/darthcloud/BlueRetro/releases/tag/v24.04'>[24.04] 2024-04-24</a>.</em>

**Comment by patrikmckane** [comment]
Im curious, why the version jump? from 1.9.2 to 24.04?

**Reply by CLHatch** [reply]
My guess would be he's switched to a date-based versioning system.  `24.04` likely either means the release in the fourth month in 2024, or it means the 4th release in 2024.

**Reply by darthcloud** [reply]
I ditched the meaning less Semantic versioning for Calendar versioning. So YY.MM. That make it easier to call what the upcoming version will be.
